src/sentry/api/endpoints/notifications/notification_actions_details.py

```diff
@@ -0,0 +1,60 @@
+from rest_framework import status
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.organization_flag import FlaggedOrganizationEndpoint
+from sentry.api.serializers import serialize
+from sentry.api.serializers.rest_framework.notification_action import NotificationActionSerializer
+from sentry.models.notificationaction import NotificationAction
+from sentry.models.organization import Organization
+
+
+@region_silo_endpoint
+class NotificationActionsDetailsEndpoint(FlaggedOrganizationEndpoint):
+    feature_flags = ["organizations:notification-actions"]
+
+    def get(self, request: Request, organization: Organization, action_id: int) -> Response:
+        try:
+            action = NotificationAction.objects.get(
+                id=action_id,
+                organization_id=organization.id,
+            )
+        except NotificationAction.DoesNotExist:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+        return Response(serialize(action, request.user))
+
+    def put(self, request: Request, organization: Organization, action_id: int) -> Response:
+        try:
+            action = NotificationAction.objects.get(
+                id=action_id,
+                organization_id=organization.id,
+            )
+        except NotificationAction.DoesNotExist:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+        serializer = NotificationActionSerializer(
+            instance=action,
+            context={
+                "access": request.access,
+                "organization": organization,
+                "user": request.user,
```

Comment:
are you using `user` or `access` in the serializer?

Reply:
Just double checked, I don't need request.user so I can remove that, but `request.access` was added since the ProjectField being used to serialize the incoming projects checks for permissions with the `access` context here.